src/main/java/io/dockstore/provision/DOSPlugin.java

```diff
@@ -49,24 +52,58 @@ public void stop() {
     @Extension
     public static class DOSPreProvision implements PreProvisionInterface {
 
-        DOSPluginUtil dosPluginUtil = new DOSPluginUtil();
         static final Set<String> SCHEME = new HashSet<>(Lists.newArrayList("dos"));
+        static final String SCHEME_PREFERENCE = "scheme-preference";
+        static final int GET_SCHEME = 0;
+        private Map<String, String> config;
+
+        DOSPluginUtil dosPluginUtil = new DOSPluginUtil();
+        List<String> preferredSchemes = new ArrayList<>();
+
+        @Override
+        public void setConfiguration(Map<String, String> map) {
+            this.config = map;
+
+            if (this.config.containsKey(SCHEME_PREFERENCE)) {
+                this.preferredSchemes = Arrays.asList(this.config.get(SCHEME_PREFERENCE).trim().split(",\\s"));
```

Comment:
👍 That's clever using the regex to remove trailing spaces.

However, I don't think it will handle the case where there is no space after a comma, nor if there is more than one space. I think it needs to be `",\\s*"`.